crates/turbo-tasks-memory/src/task.rs

```diff
@@ -620,16 +898,96 @@ impl Task {
         schedule_task
     }
 
+    fn scopes_dirty_or_active<const LIKELY_ACTIVE: bool, const INCREMENT_UNFINISHED: bool>(
```

Comment:
These should be different functions. Calling these functions is otherwise very obscure. Unless you pass in enum variants, like so:

 ```
self.scopes_dirty_or_active::<ActiveLikelihood::Likely, UnfinishedBehavior::Increment>
```

But honestly if this can be made into one, two, three, or four functions, I think I'd prefer it to using const bools for this.